### DISCUSSION TITLE
Start H264 stream without internet?  Possible?

### DISCUSSION THREAD

**Discussion by endertable** [gtxaspec/wz_mini_hacks#137] [Q&A]
Is there a way to have iCamera start the H264 stream without needing the internet?  The idea from @mrlt8 on creating the /tmp/aws_iot/rooCAok file did get iCamera to startup the kvs_stream app but it doesn't seem to start up the video stream.  I am sure the binaries are trying to reach Wyze's api server first.  I'm need about 4 cameras off the grid but I need the H264 stream vs RTSP.  You can tell when iCamera is happy when you run the following command:

netstat -anp | grep udp | grep :5555

and get an output like the following:

udp        0      0 0.0.0.0:5555            0.0.0.0:*                 {PID}/iCamera

If you leave out the "grep :5555" you should also get a few more ports which will be the actual H264 streaming ports for local and WebRTC.  

Things I have done:

1.  touch /tmp/aws_iot/rooCAok   (even populated the directory with some of the other files, before and after it started)
2.  add www.google.com and google.com to /etc/hosts
3.  add clock.fmt.he.net to /etc/hosts   (for purpose of #4)
4.  ran ntpd server on cam so that timesync shuts up and is happy
5  added api.wyzecam.com to /etc/hosts   (iCamera *sometimes* get pissed and reboots the cam after a while, hehe, but it gets the last laugh because it never starts the stream)

Not sure if this is possible without having to hook something inside of binaries or if there is some other dependency I am missing.

Thanks

**Comment by gtxaspec** [comment]
@endertable what is generating the h264 stream if it's not v4l2rtspserver?  How are you connecting to this h264 stream?  Are you referring to the P2P API video streams?

**Reply by endertable** [reply]
HI, yes it could actually be the P2P API video streams.  They are the ones that the Wyze app uses locally and remotely.  It seems iCamera, or whatever binary, doesn't start them until it has made successful internet access.  Even FiveLeaves H264 won't work, because it counts on those streams as well, I guess.  
I can start the cams using my phone as an access point so they initialize, with the internet, but if they reboot while I am gone, they won't be recorded by the DVR anymore.

**Reply by gtxaspec** [reply]
whats the difference between that stream and the rtsp one? is there a specific use case for it? just wondering

**Comment by gtxaspec** [comment]
if you are looking to stream the h264 stream, can't you do it using ffmpeg?

on the camera, ` ffmpeg -f v4l2 -input_format h264 -i /dev/video1 -vcodec copy -an -f rtp rtp://224.0.1.2:5001` 

this is a multicast example, then you can capture it or play elsewhere:

`ffplay -protocol_whitelist rtp,udp,file -i ~/test.sdp` (the sdp file is generated by ffmpeg on the command line)

test.sdp:
```
v=0
o=- 0 0 IN IP4 127.0.0.1
s=No Name
c=IN IP4 224.0.1.2
t=0 0
a=tool:libavformat 58.76.100
m=video 5001 RTP/AVP 96
a=rtpmap:96 H264/90000
a=fmtp:96 packetization-mode=1
```

**Reply by gtxaspec** [reply]
actually this probably isn't what you need since it's being encapsulated in rtp
